corehq/apps/translations/tests/test_transifex_integration.py

```diff
@@ -43,19 +61,81 @@ def test_auth_setup(self):
+        self.assertEqual(request.qs[expected_filter], [value])
 
     def test_list_objects(self):
-        ...
+        cls = self.tfx_client.api.Resource
+        key = 'project'
+        value = self.tfx_client.project.id
+        with self.mocker as mocker:
+            self.tfx_client._list_objects(cls, **{key: value}).get()
+            request = mocker.last_request
+
+        expected_filter = f"filter[{key}]"
+        self.assertEqual(request.method, 'GET')
+        self.assertEqual(request.qs[expected_filter], [value])
 
     def test_create_resource(self):
-        ...
+        resource_name = "Test Resource"
+        with self.mocker as mocker:
+            self.tfx_client._create_resource(RESOURCE_SLUG, resource_name)
+            request = mocker.last_request
+
+        data = request.json()['data']
+        self.assertEqual(request.method, 'POST')
+        self.assertEqual(data['attributes']['slug'], RESOURCE_SLUG)
+        self.assertEqual(data['attributes']['name'], resource_name)
 
     def test_delete_resource(self):
-        ...
+        with self.mocker as mocker:
+            resource = self.tfx_client._get_resource(RESOURCE_SLUG)
+            self.tfx_client.delete_resource(RESOURCE_SLUG)
+            request = mocker.last_request
+
+        self.assertEqual(request.method, 'DELETE')
+        self.assertTrue(request.path.endswith(resource.id))
 
     def test_upload_content_for_resource(self):
-        ...
+        cls = self.tfx_client.api.ResourceStringsAsyncUpload
+        content = "Here is some content"
+        key = 'resource'
+        with self.mocker as mocker:
+            value = self.tfx_client._get_resource(RESOURCE_SLUG).id
+            self.tfx_client._upload_content(cls, content, **{key: value})
+            request = mocker.last_request
+
+        text = request.text
+        self.assertEqual(request.method, 'POST')
+        self.assertIn(content, text)
+        self.assertIn(key, text)
+        self.assertIn(value, text)
 
     def test_download_content_for_resource(self):
-        ...
+        cls = self.tfx_client.api.ResourceStringsAsyncDownload
+        key = 'resource'
+        with self.mocker as mocker:
+            value = self.tfx_client._get_resource(RESOURCE_SLUG).id
+            content = self.tfx_client._download_content(cls, **{key: value})
+            post_download, get_download_status, get_content = mocker.request_history[-3:]
+
+        # creates async download request
+        data = post_download.json()['data']
+        self.assertEqual(post_download.method, 'POST')
+        self.assertEqual(data['attributes'][key], value)
+
+        # checks status of download
+        self.assertEqual(get_download_status.method, 'GET')
+
+        # returns expected content
+        expected_content = self.get_file(DATA_PATH + '/get_resource_strings_async_downloads_content', 'json')
+        self.assertEqual(get_content.method, 'GET')
+        self.assertEqual(content.decode('utf-8'), expected_content)
```

Comment:
I can't tell if this assert adds any value since I'm defining both the expected content and the redirect to it (via `_handle_download_redirect`. Maybe it shows that the redirect was followed correctly? With these other API interaction tests, though, I haven't found it worthwhile to check return values that I'm already predefining like this, so I'll probably remove this assert.